### PR TITLE
Fix warnings with -Wpedantic

### PR DESCRIPTION
**Pull request by keith**
## Goal

This fixes a few small warnings that popped up when using super strict
warning flags.

image

## Design

Internally we build Bugsnag with Bazel, to fix these externally I added `-Wpedantic -Wall -Wextra` to `OTHER_CFLAGS` for testing (I did not push that).

## Tests

This shouldn't introduce any behavioral change

### Outstanding Questions

- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review
  - [ ] Release

### CODE REVIEW

File: Source/BugsnagCrashReport.m [hunk 1]
```diff
@@ -671,12 +671,9 @@ + (instancetype)errorDataFromThreads:(NSArray *)threads {
             }
             NSString *contentValue = data[@"value"];
 
-#pragma clang diagnostic push
-#pragma ide diagnostic ignored "OCDFAInspection"
```
Comment (keith) [inline Source/BugsnagCrashReport.m:675]: I couldn't find anything online on what this actually did, but the compiler warning implied that it was ignored

**Comment by keith** [comment]
If you're interested being compatible with these super strict warnings in general I'm happy to push the bit I was testing with which was something like this:

```
+OTHER_CFLAGS = $(inherited) -Wpedantic -Wall -Wextra -Wno-gnu-zero-variadic-macro-arguments -Wno-gnu-conditional-omitted-operand -Wno-unused-parameter -Wno-missing-field-initializers -Wno-gnu-statement-expression -Wno-zero-length-array
```

**Comment by keith** [comment]
🤔 the test failure here appears to be unrelated

**Comment by mattdyoung** [comment]
Hi @keith 

Thanks for the PR! We'll review and consider this in due course.

**Comment by tomlongridge** [comment]
This is now released under v6.0.1.

**Pull request closed without merging**